### DISCUSSION TITLE
combine AND and OR criteria in grid typescript

### DISCUSSION THREAD

**Discussion by teafmartins** [serenity-is/Serenity#6143] [Help] [answered]
Hello,

How can I combine AND criteria and OR criteria in grid typescript?
something like this:  departament in h.vallue AND (type=1 OR type4)

h.request.Criteria = Serenity.Criteria.and(h.request.Criteria, [["department"], 'in', [h.value]]); 
...
h.request.Criteria = Serenity.Criteria.or([["type"], '=', 1], [["type"], '=', 4]));

Thanks

**Comment by dfaruque** [comment, marked as answer]
try something like the following
```
let deptCriteria = [["department"], 'in', [h.value]];
let type1or4Criteria = Serenity.Criteria.or([["type"], '=', 1], [["type"], '=', 4]));
h.request.Criteria = Serenity.Criteria.and(h.request.Criteria, deptCriteria , type1or4criteria );
```

**Reply by teafmartins** [reply]
Perfect! Thanks!
